src/eth.rs

```diff
@@ -72,6 +72,44 @@ pub enum RmiiEthChipset {
     KSZ80XX,
 }
 
+#[cfg(all(esp32, esp_idf_eth_use_esp32_emac))]
+pub enum RmiiClockConfig {
+    Input(gpio::Gpio0<gpio::Unknown>),
+    #[cfg(not(esp_idf_version = "4.3"))]
+    OutputGpio0(gpio::Gpio0<gpio::Unknown>),
+    /// This according to ESP-IDF is for "testing" only
+    #[cfg(not(esp_idf_version = "4.3"))]
+    OutputGpio16(gpio::Gpio16<gpio::Unknown>),
+    #[cfg(not(esp_idf_version = "4.3"))]
+    OutputInvertedGpio17(gpio::Gpio17<gpio::Unknown>),
+}
+
+#[cfg(all(esp32, esp_idf_eth_use_esp32_emac, not(esp_idf_version = "4.3")))]
+impl RmiiClockConfig {
+    fn eth_mac_clock_config(&self) -> eth_mac_clock_config_t {
+        let rmii = match self {
+            Self::Input(_) => eth_mac_clock_config_t__bindgen_ty_2 {
+                clock_mode: emac_rmii_clock_mode_t_EMAC_CLK_DEFAULT,
```

Comment:
I think this should **not** be `emac_rmii_clock_mode_t_EMAC_CLK_DEFAULT`, but `emac_rmii_clock_mode_t_EMAC_CLK_EXT_IN`. With `emac_rmii_clock_mode_t_EMAC_CLK_DEFAULT`, the configuration settings specified in `sdkconfig.defaults` will be used instead, and we do not want this, as agreed earlier.

Check [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_eth.html) for more info.

Reply:
Good catch! Thanks :)